apps/labs/posts/pandas-resample.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Why you probably shouldn't be using `df.resample('M')`"
+authors: [marco-gorelli]
+published: June 17, 2024
+description: 'Learn what you should be using instead'
+category: [PyData ecosystem]
+featuredImage:
+  src: /posts/pandas-resample/pandas-resample.png
+  alt: 'Image of Drake not looking impressed at `resample("M")`, and preferring `resample("MS")`'
+hero:
+  imageSrc: /posts/pandas-resample/pandas-logo-wide-field.png
+  imageAlt: 'Pandas logo'
+---
+
+Suppose you have a Dataframe `df` containing two columns:
+
+- date;
+- price.
+
+You can want to find the average price per month, and are using
+pandas. Looking around online, you might find a solution such as
+
+```python
+df.resample('M', on='date')['price'].mean()
+# or, since pandas 2.1: df.resample('ME', on='date')['price'].mean()
+```
+
+which divides the data into monthly buckets and finds the mean value of
+`'price'` for each one.
+
+There's nothing wrong with that - however, in this post, I'd like to make the
+case that you should almost certainly replace it with
+
+```python
+df.resample('MS', on='date')['price'].mean()
+```
+
+Let's learn why!
+
+## What happens when we resample by 'M' (or 'ME')
+
+Before we begin, let's talk about `'M'`. You'd be forgiven for thinking that it stands for "month".
+But actually, looking at the [pandas docs](https://pandas.pydata.org/pandas-docs/version/2.1/user_guide/timeseries.html#offset-aliases), we see that it stands for "month end"! Indeed in pandas 2.1 it was deprecated and
+renamed to `'ME'` to make its meaning clearer.
```

Comment:
You mention that `ME` means `Month End` here but I don't think you define what `MS` stands for anywhere.